c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -129,19 +129,25 @@ using stream_set = ska::flat_hash_set<cuda::CUDAStream>;
 
 using StatTypes = std::array<bool, static_cast<size_t>(StatType::NUM_TYPES)>;
 
-void update_stat(Stat& stat, int64_t amount) {
-  stat.current += amount;
+void increase_stat(Stat& stat, size_t amount) {
+  stat.current += static_cast<int64_t>(amount);
+  stat.peak = std::max(stat.current, stat.peak);
+  stat.allocated += static_cast<int64_t>(amount);
+}
 
+void decrease_stat(Stat& stat, size_t amount) {
+  stat.current -= static_cast<int64_t>(amount);
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
       stat.current >= 0,
       "Negative tracked stat in CUDA allocator (likely logic error).");
+  stat.freed += static_cast<int64_t>(amount);
+}
 
-  stat.peak = std::max(stat.current, stat.peak);
+void update_stat(Stat& stat, int64_t amount) {
```

Comment:
Why do you preserve this one? Do we still use it?

Reply:
Removed.